src/main/scala/org/broadinstitute/dsde/agora/server/dataaccess/permissions/PermissionsClient.scala

```diff
@@ -45,42 +45,55 @@ trait PermissionsClient {
 
     addUserIfNotInDatabase(userEmail)
 
+    // Construct query to get permissions
     val permissionsQuery = for {
-      user <- users if user.email === userEmail
-      entity <- entities if entity.alias === alias(agoraEntity)
-      permission <- permissions if permission.entityID === entity.id &&
-                                    permission.userID === user.id
+      user <- users
+        .filter(_.email === userEmail)
+        .result
+        .head
+
+      entity <- entities
+        .filter(_.alias === alias(agoraEntity))
+        .result
+        .head
+
+      permission <- permissions
+        .filter(p => p.entityID === entity.id && p.userID === user.id)
+        .result
+        .headOption
     } yield permission
 
-    val permissionsFuture = db.run(permissionsQuery.result)
-
-    val agoraPermissions = permissionsFuture.map { permissionsResult =>
-      val permissionResult = permissionsResult.headOption
+    // run query
+    try {
+      val permissionResult = Await.result(db.run(permissionsQuery), timeout)
 
       if (permissionResult.isEmpty)
         AgoraPermissions(Nothing)
       else
         AgoraPermissions(permissionResult.get.roles)
 
-    } recover {
-      case ex: Throwable => throw new PermissionNotFoundException(s"Could not get permissions", ex)
+    } catch {
+      case ex: Throwable => throw new PermissionNotFoundException(s"Could not get permission", ex)
     }
 
-    Await.result(agoraPermissions, timeout)
   }
 
   def listPermissions(agoraEntity: AgoraEntity): Seq[AccessControl] = {
+    // Construct query
     val permissionsQuery = for {
       entity <- entities if entity.alias === alias(agoraEntity)
       _permissions <- permissions if _permissions.entityID === entity.id
       user <- users if user.id === _permissions.userID
     } yield (user.email, _permissions.roles)
```

Comment:
honestly, its just a lack on Slick knowledge. This query is different because it returns a sequence of permissions, while the other queries return an individual permission. I would guess there is a way to expand / simplify the query, but I'm just not sure how to do it right now.